contoso_manufacturing/developer/webapp-decode/app.py

```diff
@@ -178,9 +179,13 @@ def gen_frames(video_name):
               latest_choice_detector = init_pose_estimator()
 
     while video_name != "":
-        processed_frame = latest_choice_detector.run()
-        if processed_frame is not None:
-            ret, buffer = cv2.imencode('.jpg', processed_frame)
+        while latest_choice_detector.postprocessed_frames_queue.empty():
+            print("Waiting for postprocessed frames...")
+            time.sleep(0.1)
```

Comment:
100ms might be too long to wait, if total processing time per frame could be as low as 33ms (30 FPS)?  
- Unless JPG encoding is the slowest step (which isn't the case right now at least) then some occasional queue misses are expected.  
- Waiting too long here could make the output jittery.. the output will occasionally 'freeze' for 100ms and then playback of the next several frames will be too fast as the JPG encoding 'catches' up with what has been added to the queue in the meantime.
Reducing this to say 2-5ms or some other small fraction of the expected cycle time should help to smooth out occasional misses where the 4 threads are all completing work at different rates.   (Both here and in the relevant sleep commands in yolov8.py)

Reply:
I meant to keep this 10ms like other waits in yolov8.py but mistakenly put 100ms instead. Changed every wait to 5ms to include more checks.